Feature/disable check

absinth84 opened the pull request:
Added a Flag (timeout = -2 ) to disable deploy check after service update.

absinth84 commented:
Why the automatic test with Travis-ci fails?

fabfuel commented:
Hi @absinth84, this was related to the the new version 7.0.0 of Click.
I fixed this in the `develop` and `master` branches, please update your branch, then the build should pass.

fabfuel commented on README.rst:
```diff
@@ -264,6 +264,8 @@ The timeout error means, that AWS ECS takes longer for the full deployment cycle
 
 You can increase the time (in seconds) to wait for finishing the deployment via the ``--timeout`` parameter. This time includes the full cycle of stopping all old containers and (re)starting all new containers. Different stacks require different timeout values, the default is 300 seconds.
 
+If you set '--timeout = -2' check deployment is skipped
```
This syntax is not correct, it needs to be `--timeout -2`

```suggestion
To completely skip the validation of the successful deployment, you can disable the timeout by setting::
    
    --timeout -2
```

fabfuel commented on ecs_deploy/cli.py:
```diff
@@ -239,14 +239,15 @@ def deploy_task_definition(deployment, task_definition, title, success_message,
-        failure_message=failure_message,
-        ignore_warnings=ignore_warnings
-    )
+    if(timeout != -2):
```
Why `-2`?

fabfuel commented:
Hey, 
as there were no tests added yet, i implemented the "fire and forget" deployment in #81 (via `--timeout -1`).
Best, Fabian

absinth84 commented:
Hey Fabian,
Sorry for the late but i was in holiday..
Thank you!

I use ecs deploy in parallel mode from jenkins (15 - 20 services at the
same time x some clusters) and to not overcall the aws api and to not
receive to many call i added also a random sleep between 10 to 40 sec
before describe_services call.

Best,

Paolo Cerletti

Il giorno sab 9 feb 2019 alle ore 12:51 Fabian Fuelling <
notifications@github.com> ha scritto:

> Hey,
> as there were no tests added yet, i implemented the "fire and forget"
> deployment in #81 <https://github.com/fabfuel/ecs-deploy/pull/81> (via --timeout
> -1).
> Best, Fabian
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/fabfuel/ecs-deploy/pull/68#issuecomment-462037934>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/APRp7h8PgijShoQaDlYvhDX8XoqcrGr6ks5vLrZVgaJpZM4X79py>
> .
>
